bindings/go/src/examples/perftest/perftest.go

```diff
@@ -0,0 +1,397 @@
+		backwardEp, _ := perfTest.perThreadWorkers[0].NewEndpoint((&UcpEpParams{}).
+			SetConnRequest(connRequest).SetPeerErrorHandling())
+
+		// Current connection establishment requires to create backward ep or reject connection.
+		perfTest.backwardEps = append(perfTest.backwardEps, backwardEp)
+		fmt.Println("Got connection starting benchmark")
+	})
+
+	perfTest.listener, err = perfTest.perThreadWorkers[0].NewListener(listenerParams)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Started receiver listener on address: %v \n", addr)
+	return nil
+}
+
+func progressWorker(i int) {
+	perfTest.perThreadWorkers[i].Progress()
+}
+
+func submitTagRecvRequest(tid uint) {
+	requestParams := &UcpRequestParams{}
+	requestParams.SetMemType(perfTestParams.memType).SetCallback(func(request *UcpRequest, status UcsStatus, tagInfo *UcpTagRecvInfo) {
+		perfTest.numCompletedRequests += 1
+		request.Close()
+		submitTagRecvRequest(tid)
+	})
+	perfTest.perThreadWorkers[0].RecvTagNonBlocking(getAddressOffsetForThread(tid), perfTestParams.messageSize, uint64(tid), uint64(tid), requestParams)
+}
+
+func close() {
+	for _, backwardEp := range perfTest.backwardEps {
+		backwardEp.CloseNonBlockingForce(nil)
+	}
+
+	for _, ep := range perfTest.eps {
+		ep.CloseNonBlockingForce(nil)
+	}
+
+	if perfTest.listener != nil {
+		perfTest.listener.Close()
+	}
+
+	for _, worker := range perfTest.perThreadWorkers {
+		worker.Close()
+	}
+
+	if perfTest.memory != nil {
+		perfTest.memory.Close()
+	}
+
+	if perfTest.context != nil {
+		perfTest.context.Close()
+	}
+}
+
+func startServer() error {
+	initContext()
+	if err := initMemory(); err != nil {
+		return err
+	}
+	perfTest.perThreadWorkers = make([]*UcpWorker, 1)
+	initWorker(0)
+	if err := initListener(); err != nil {
+		return err
+	}
+
+	if perfTestParams.protocol == "am" {
+		for t := uint(0); t < perfTestParams.numThreads; t += 1 {
+			tid := t
+			perfTest.perThreadWorkers[0].SetAmRecvHandler(t, UCP_AM_FLAG_WHOLE_MSG,
+				func(header unsafe.Pointer, headerSize uint64, data *UcpAmData, replyEp *UcpEp) UcsStatus {
+					if data.IsDataValid() {
+						perfTest.numCompletedRequests += 1
+						return UCS_OK
+					} else {
+						data.Receive(getAddressOffsetForThread(tid), perfTestParams.messageSize,
+							(&UcpRequestParams{}).SetMemType(perfTestParams.memType).SetCallback(func(request *UcpRequest, status UcsStatus, length uint64) {
+								perfTest.numCompletedRequests += 1
+								request.Close()
+							}))
+
+						return UCS_OK
+					}
+				})
+		}
+	}
+
+	for i := uint(0); i < perfTestParams.numIterations; i += 1 {
+		perfTest.numCompletedRequests = 0
+		// For tag need to submit every time recv request
+		if perfTestParams.protocol == "tag" {
+			for t := uint(0); t < perfTestParams.numThreads; t += 1 {
+				submitTagRecvRequest(t)
+			}
+		}
+
+		for perfTest.numCompletedRequests < perfTestParams.numThreads {
+			progressWorker(0)
+		}
+	}
+	close()
+	return nil
+}
+
+func iterPerThread(i uint, t uint) {
```

Comment:
can we add comment, or specific name, to distinguish client and server functions?
for example: clientThreadDoIter